parcels/_index_search.py

```diff
@@ -0,0 +1,336 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+from parcels._typing import (
+    GridIndexingType,
+    InterpMethodOption,
+)
+from parcels.tools.statuscodes import (
+    FieldOutOfBoundError,
+    FieldOutOfBoundSurfaceError,
+    _raise_field_out_of_bound_error,
+    _raise_field_out_of_bound_surface_error,
+    _raise_field_sampling_error,
+)
+
+from .grid import GridType
+
+if TYPE_CHECKING:
+    from .field import Field
+    from .grid import Grid
+
+
+def search_indices_vertical_z(grid: Grid, gridindexingtype: GridIndexingType, z: float):
```

Comment:
Would these methods not be private? Do we expect users to call `search_indices_vertical_z` directly?

Reply:
The methods are in a private module `_index_search.py`

Reply:
Ah ok, so if the module is private then by default all methods are private too. That actually makes sense, now that I think about it ;-)